### PR TITLE
Add script for docs preprocessing for static site

### PR DESCRIPTION
**Pull request by BarkingBad**

### CODE REVIEW

File: project/scripts/processDocs.sc [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import java.nio.file.{ FileSystems, Files, Path }
+import java.io.PrintStream
+import scala.io.{ Codec, Source }
+import scala.collection.JavaConverters._
+
+val inputDir = FileSystems.getDefault.getPath("docs-for-scalalang")
```
Comment (romanowski) [inline project/scripts/processDocs.sc:6]: I think we keep docs in `docs` dir and process them into `docs-for-dotty-page` dir to avoid confusion.

**Comment by BarkingBad** [comment]
To review the PR it's good to filter out md files like this:
obraz
The only interesting md file is `docs/README.md` file explaining why we have two types of docs there and how to build documentation. The goal is to port all the files to the `docs.scala-lang` so the number of whitelisted (or later blacklisted) directories should change and eventually all files will be preprocessed.

**Comment by BarkingBad** [comment]
https://scala3doc.virtuslab.com/pr-move-docs/scala3/docs/index.html

From static site point of view everything is working correctly.

**Comment by BarkingBad** [comment]
If you copy the `docs/docs/reference` into `_scala3-reference` and `docs/docs/usage/scaladoc` into `_overviews/scala3-scaladoc` you can generate the `docs.scala-lang` page with no issues (actually these files has been backported from `docs.scala-lang`). Of course these files will be preprocessed, but for now it is sufficient to prove that they work with docs.scala-lang.

File: project/scripts/processDocs.sc [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import java.nio.file.{ FileSystems, Files, Path, StandardCopyOption }
+import java.io.PrintStream
+import scala.io.{ Codec, Source }
+import scala.collection.JavaConverters._
+
+val inputDir = FileSystems.getDefault.getPath("docs")
```
[the remark this answers is not in the document]
Comment (BarkingBad) [reply project/scripts/processDocs.sc]: Added in `docs/README.md`

**Pull request merged**